ocp_resources/data_science_cluster.py

```diff
@@ -0,0 +1,88 @@
+from ocp_resources.resource import Resource
+
+
+class DataScienceCluster(Resource):
+    """
+    DataScienceCluster Resource for RHOAI operator
+
+    https://access.redhat.com/documentation/en-us/red_hat_openshift_ai_self-managed/2.5/html/installing_and_uninstalling_openshift_ai_self-managed/index
+    """
+
+    api_group = Resource.ApiGroup.DATASCIENCE_CLUSTER
+
+    class ManagementState:
+        """
+        Supported Management States of components of DataScienceCluster.
+
+        Managed: The Operator actively manages the component, installs it, and tries to keep it active.
+        The Operator will upgrade the component only if it is safe to do so.
+
+        Removed: The Operator actively manages the component but does not install it. If the component is
+        already installed, the Operator will try to remove it.
+        """
+
+        MANAGED = "Managed"
+        REMOVED = "Removed"
+
+    def __init__(
+        self,
+        name=None,
+        client=None,
+        yaml_file=None,
+        codeflare_state=ManagementState.REMOVED,
+        dashboard_state=ManagementState.MANAGED,
+        datasciencepipelines_state=ManagementState.REMOVED,
+        kserve_state=ManagementState.REMOVED,
+        modelmeshserving_state=ManagementState.REMOVED,
+        ray_state=ManagementState.REMOVED,
+        trustyai_state=ManagementState.REMOVED,
+        workbenches_state=ManagementState.REMOVED,
```

Comment:
do not set defaults; either the api has defaults and if the user does not provide - do not set in the code or they are mandatory and the user must provide them

Reply:
checking if there are defaults

Reply:
If there is no values passed for either one of the components, then DSC still getting created, without ManagemantState, see example. Removing the default values
```
spec:
  components:
    codeflare:
      devFlags: {}
      managementState: Removed
    dashboard:
      devFlags: {}
      managementState: Managed
    datasciencepipelines:
      devFlags: {}
      managementState: Removed
    kserve:
      devFlags: {}
      managementState: Removed
    modelmeshserving:
      devFlags: {}
      managementState: Removed
    ray:
      devFlags: {}
      managementState: Removed
    trustyai:
      devFlags: {}
      managementState: Removed
    workbenches:  ## >>>> NO State passed for this component <<<<<<
      devFlags: {}
```